src/main/java/edu/bu/metcs673/trackr/service/impl/UserServiceImpl.java

```diff
@@ -25,23 +32,55 @@ public class UserServiceImpl implements UserService {
 	@Autowired
 	private UserRepository userRepository;
 
-	@Override
-	public List<User> retrieveAllUsers() {
-		// TODO Auto-generated method stub
-		return null;
-	}
-
 	@Override
 	public User findUserById(long id) {
 		log.info("Searching for record id: " + id);
-		User user = userRepository.findById(id);
+		User user = userRepository.findById(id).get();
 		return user;
 	}
 
 	@Override
-	public void createUser(String username, String password, String email) {
-		// TODO Auto-generated method stub
+	public ResponseEntity<GenericApiResponse> createUser(User userInput) {
+		// validate parameters
+		validateParameters(userInput);
```

Comment:
Now back to my comment about builder vs bean. I would think that validation would happen when you're building the User model hence it would be part of the model build method (or constructor). Everything about a User validation would be encapsulated in the User model.

Reply:
Yeah I'll refactor and use an annotation based validation for the User object.  The validation will occur at the Controller level and I updated the Exception handler to look for `MethodArgumentNotValidException`. 

Refining the validate params method to only include stuff not covered by annotations such as the unique username check.

Reply:
Ok sounds good.